### PR TITLE
fix(license-scan): classify marker-excluded deps via static fallback

### LINKED ISSUE #1258: Follow-up from #1219: 1 item(s) (core)
_Consolidated follow-up from implementation of #1219._

Each section below lists scope-checked follow-up items discovered during implementation. Items are restricted to core library defects, security, safety hazards, or critical functional bugs.

### license-scan never classifies tomli/tzdata (single-Py Linux CI gap)

The license-scan job (.github/workflows/security.yml) runs only python-version 3.13 on ubuntu-24.04. scripts/check_license_compatibility.py:scan() skips-with-note any in-scope dep whose marker excludes the current interpreter (the `installable_now=False` branch). Result: `tomli` (pyproject.toml gated python_version<'3.11') and `tzdata` (pyproject.toml gated platform_system=='Windows') are in the distributed scope set but are NEVER license-classified by any CI row — a permanent silent coverage hole in the gate that #1219 built to eliminate silent holes. Both are MIT today so no current violation, but a future transitive upgrade of either would not be caught. Sketch fix: either (a) add a second license-scan matrix leg (python-version 3.10 to cover tomli; document tzdata as Windows-unreachable-on-Linux) so every distributed dep is classified by at least one row, or (b) add a static fallback license map for markered-out deps (tomli->MIT, tzdata->MIT per NOTICE) so resolve_license can classify them without an install. Lock with a test asserting the union of classified deps across CI rows covers the full distributed_requirements set.

---

_The implementer also considered the items below and rejected them as out of scope; they are recorded in the PR body, not filed as separate issues._

### PR DESCRIPTION
**Pull request by mvillmow**
Add `STATIC_FALLBACK_LICENSES` to classify `tomli` and `tzdata` on the Python 3.13/Linux CI leg where their markers exclude them from installation. Previously these deps were skipped with a misleading note; now they resolve to their SPDX license from NOTICE, and unknown marker-excluded deps exit(2) instead of silently passing.

**Changes:**
- `scripts/check_license_compatibility.py`: add `STATIC_FALLBACK_LICENSES` dict (MIT for tomli, Apache-2.0 for tzdata); replace silent-skip branch with fallback-or-exit(2) logic; update docstring with new FAILS LOUDLY bullet
- `tests/unit/scripts/test_check_license_compatibility.py`: rename contradicted test; add `TestStaticFallback` class with 6 tests including two staleness-mitigation cross-checks (NOTICE value bound to per-package lines, real metadata when dep is installed)

Closes #1258

### CODE REVIEW

File: tests/unit/scripts/test_check_license_compatibility.py [hunk 1]
```diff
@@ -257,3 +258,102 @@ def test_pygithub_lgpl_fixture_passes(self, tmp_path):
...
+    """STATIC_FALLBACK_LICENSES classifies marker-excluded deps instead of skipping.
+
+    Staleness mitigations:
+    - test_static_values_match_installed_metadata: cross-checks against real
+      importlib.metadata when the dep IS installable (catches license drift).
+    - test_static_values_match_notice: cross-checks against NOTICE (the
+      authoritative human-readable analysis per the script's own docstring).
+    """
+
+    def test_tomli_fallback_classifies_as_compatible(self):
```
Comment (mvillmow) [inline tests/unit/scripts/test_check_license_compatibility.py:273]: P4/DRY: `test_tomli_fallback_classifies_as_compatible` is a verbatim duplicate of `test_uninstalled_other_python_dep_with_fallback_classifies_not_fails` (line 143): identical patches, identical assertion, identical package. Since the renamed test in `TestLoudFailure` already covers 'known fallback → compatible, not a hole', this test adds zero new coverage. Either remove it and let the `TestLoudFailure` test carry the contract, or differentiate it (e.g. assert the printed note contains the package name, or assert `ids == ["MIT"]` from a spy).

File: scripts/check_license_compatibility.py [hunk 2]
```diff
@@ -263,13 +277,22 @@ def scan(records: dict[str, dict] | None = None) -> list[tuple[str, list[str]]]:
...
             print(
-                f"note: skipping {pkg!r} — distributed only under another "
-                "Python/platform (marker excludes this interpreter); it is "
-                "classified on the matching CI matrix row.",
+                f"note: {pkg!r} excluded by marker in this env; "
+                f"classifying via static fallback from NOTICE: {fallback}",
                 file=sys.stderr,
             )
-            continue
+            ids = fallback
```
Comment (mvillmow) [inline scripts/check_license_compatibility.py:296]: P7/POLA: The `scan()` docstring (line 264, pre-existing) now contradicts the behavior introduced by this diff. It says marker-excluded deps "are skipped with a note, not treated as a hole" — but lines 280-295 (new in this diff) either classify them via `STATIC_FALLBACK_LICENSES` or `exit(2)`. A caller reading only the docstring will have incorrect expectations. Update the docstring to: "Deps absent only because their marker excludes the current interpreter are classified via `STATIC_FALLBACK_LICENSES`, or exit 2 if the dep has no fallback entry."

**Review by mvillmow — COMMENTED** [review]
Implementation is correct and complete: fallback dict, exit(2) for unknown deps, and NOTICE cross-check are all sound. Two minor issues: duplicate tomli test and stale scan() docstring.

File: tests/unit/scripts/test_check_license_compatibility.py (hunk 1 above)
Comment (mvillmow) [reply tests/unit/scripts/test_check_license_compatibility.py:273]: Resolved by the automation reviewer after a GO implementation review.

File: scripts/check_license_compatibility.py (hunk 2 above)
Comment (mvillmow) [reply scripts/check_license_compatibility.py:296]: Resolved by the automation reviewer after a GO implementation review.
